src/main/java/org/ngafid/flights/process/CSVFileProcessor.java

```diff
@@ -46,105 +54,408 @@ private CSVFileProcessor(Connection connection, byte[] bytes, String filename, P
+                    stringTimeSeries.put("Lcl Time", new StringTimeSeries("Lcl Time", "HH:mm:ss", localDateTimeResult.getLocalTimes()));
+                    stringTimeSeries.put("UTCOfst", new StringTimeSeries("UTCOfst", "hh:mm", localDateTimeResult.getUtcOffsets()));
+                }
+
+                // Populate doubleTimeSeries and stringTimeSeries
+                int startIndex = isG5FlightRecorder ? 2 : 0; // Skip first two columns (UTC date/time) if it is G5 flight recorder
+                for (int j = startIndex; j < columns.size(); j++) {
+                    var column = columns.get(j);
+                    var name = headers.get(j);
+                    var dataType = dataTypes.get(j);
+
+                    try {
+                        Double.parseDouble(column.get(0));
+                        doubleTimeSeries.put(name, new DoubleTimeSeries(name, dataType, column));
+                    } catch (NumberFormatException e) {
+                        stringTimeSeries.put(name, new StringTimeSeries(name, dataType, column));
+                    }
+                }
+
+                String md5Hash = calculateMd5Hash(flightRows);
+
+                // Build and add a flight
+                FlightMeta newMeta = new FlightMeta(meta);
+                newMeta.setMd5Hash(md5Hash);
+
+                FlightBuilder builder;
+                if (isG5FlightRecorder) {
+                    builder = new G5FlightBuilder(newMeta, doubleTimeSeries, stringTimeSeries);
+                } else {
+                    builder = new CSVFlightBuilder(newMeta, doubleTimeSeries, stringTimeSeries);
+                }
+
+                flightBuilders.add(builder);
+                doubleTimeSeries.clear();
+                stringTimeSeries.clear();
             }
+        } catch (IOException | FatalFlightFileException | CsvException e) {
+            throw new FlightProcessingException(e);
+        } catch (SQLException e) {
+            String errorMessage = "SQL error occurred while processing the flight data for file: " + filename;
+            throw new FlightProcessingException(errorMessage, e);
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e);
+        }
 
-            for (int i = 0; i < columns.size(); i++) {
-                var column = columns.get(i);
-                var name = headers.get(i);
-                var dataType = dataTypes.get(i);
+        LOG.info("Parse method end. Returning " + flightBuilders.size() + " flight builders.");
+        return flightBuilders.stream();
+    }
+
+    /**
+     *  Ands derived CSV files to the uploads folder
+     * @param splitIndices
+     * @param rows
+     * @param headerLines
+     * @param filename
+     * @throws IOException
+     * @throws SQLException
+     */
+    private void addDerivedFileToUploadFolder(List<Integer> splitIndices, List<String[]> rows, List<String> headerLines, String filename) throws IOException, SQLException {
```

Comment:
Do you need to write this to disk? I figure it can be added directly to the zip. Doing this will also leave junk data on the filesystem if not removed properly.